### ISSUE TITLE
[LUA] Allow AddStaticObject to replace anything

### ISSUE THREAD

**Issue by kubaau** [Return-To-The-Roots/s25client#1692]
S2 RTX event `!MET_SET_MAP_ELEMENT` was almost always used to activate gates (replace existing gate with activated one), but there is at least one interesting case where it was used to destroy buildings. In 2NDpaign chapter 3 occupying a certain position used `!MET_SET_MAP_ELEMENT` to place a rock where the player's HQ (and harbor) were. This essentially destroyed these buildings (although there was no fire, no-one left and borders were not recalculated):

![Zrzut ekranu 2024-07-29 001013](https://github.com/user-attachments/assets/430ffbf2-2efd-4b70-a8a0-bc6fb798a4b5)
![Zrzut ekranu 2024-07-29 001036](https://github.com/user-attachments/assets/cedaa17e-0cf7-48eb-a1e1-f52b26dfebfd)

A similar effect can be achieved in RttR by modifying LUA's `AddStaticObject` to replace whatever is there with no regard:
![Zrzut ekranu 2024-07-29 002102](https://github.com/user-attachments/assets/717e8dc4-e30b-4f78-a2fd-6cb725e3fa64)

**Comment by Spikeone** [comment]
This should be an extra parameter and should probably destroy any resulting invalid objects (e.g. in this case the road segment and border).

**Comment by Flamefire** [comment]
This would still modify the border when a recalculation happens as the building holding the border doesn't exist any more and I don't see how to easily change that. So what exactly should this do and why?

**Comment by kubaau** [comment]
I don't think this should necessarily mirror the S2 implementation because these last pictures are rather confusing. In S2 it is not even obvious what happened because you don't get folk running around as if an HQ/harbor was destroyed, just a silent rock is placed with a road and a flag sticking out of it :D The intention of this ticket is to have the option to replace any object. What follows is up for discussion :)

To get the strangle-looking implementation above I commented out a `DestroyNO` call in `LuaWorld::AddStaticObject` but leaving it in results in a more sane result:
![Zrzut ekranu 2024-07-29 123432](https://github.com/user-attachments/assets/e84ca817-ccb0-4aa9-9151-2e388df14af0)
